app/config.py

```diff
@@ -1,13 +1,35 @@
-import yaml
 import os
 from pathlib import Path
 import collections
+from deepdiff import DeepDiff
+import yaml
+from pprint import pprint
 
 from ..log import make_log
 
 log = make_log('baselayer')
 
 
+def ensure_yaml_structure_matches_defaults(config_path, defaults_config_path):
+    # Borrowed from https://github.com/dmitryduev/kowalski/blob/master/kowalski.py
+    # check contents of config WRT config.yaml.defaults
+    with open(defaults_config_path) as defaults_yaml:
+        config_defaults = yaml.load(defaults_yaml, Loader=yaml.FullLoader)
+    with open(config_path) as config_yaml:
+        config_wildcard = yaml.load(config_yaml, Loader=yaml.FullLoader)
+    deep_diff = DeepDiff(config_wildcard, config_defaults, ignore_order=True)
```

Comment:
I think we (also) want the option to check whether a config file complies with the schema.  For that case `ignore_order` must be set to False.

Reply:
@dmitryduev I presume this was written to ensure that all fields are present in the config, which is not a strict requirement for us?

Reply:
@stefanv it is WRT app.routes -- so perhaps I could change this to simply log a message if any other parts of the config don't match, but error if app.routes in config.yaml doesn't match the defaults (which will lead to missing route/component mappings and critical front-end crashes)

Reply:
@stefanv yes, that's a requirement on Kowalski, but not here, so I support @acrellin's suggestion.

Reply:
Done